test/remote/gateways/remote_securion_pay_test.rb

```diff
@@ -0,0 +1,341 @@
+    assert authorization = @gateway.authorize(@amount, @credit_card, @options)
+    assert_success authorization
+    assert !authorization.params["captured"]
+    assert_equal @options[:description], authorization.params["description"]
+    assert_equal @options[:email], authorization.params["metadata"]["email"]
+
+    assert response = @gateway.capture(@amount, authorization.authorization)
+    assert_success response
+  end
+
+  def test_failed_authorize
+    assert response = @gateway.authorize(@amount, @declined_card, @options)
+    assert_failure response
+  end
+
+  def test_partial_capture
+    assert auth = @gateway.authorize(@amount, @credit_card, @options)
+    assert_success auth
+
+    assert capture = @gateway.capture(@amount-1, auth.authorization)
+    assert_failure capture
+  end
+
+  def test_failed_capture
+    assert response = @gateway.capture(@amount, 'invalid_authorization_token')
+    assert_failure response
+    assert_match %r{Requested Charge does not exist}, response.message
+  end
+
+  def test_successful_full_refund
+    assert purchase = @gateway.purchase(@amount, @credit_card, @options)
+    assert_success purchase
+    assert purchase.authorization
+
+    assert refund = @gateway.refund(@amount, purchase.authorization)
+    assert_success refund
+
+    assert refund.params["refunded"]
+    assert_equal 0, refund.params["amount"]
+    assert_equal 1, refund.params["refunds"].size
+    assert_equal @amount, refund.params["refunds"].map{|r| r["amount"]}.sum
+
+    assert refund.authorization
+  end
+
+  def test_successful_partially_refund
+    assert purchase = @gateway.purchase(@amount, @credit_card, @options)
+    assert_success purchase
+    assert purchase.authorization
+
+    assert first_refund = @gateway.refund(@refund_amount, purchase.authorization)
+    assert_success first_refund
+
+    assert second_refund = @gateway.refund(@refund_amount, purchase.authorization)
+    assert_success second_refund
+    assert second_refund.params["refunded"]
+    assert_equal @amount - 2 * @refund_amount, second_refund.params["amount"]
+    assert_equal 2, second_refund.params["refunds"].size
+    assert_equal 2 * @refund_amount, second_refund.params["refunds"].map{|r| r["amount"]}.sum
+    assert second_refund.authorization
+  end
+
+  def test_unsuccessful_authorize_refund
+    assert response = @gateway.refund(@amount, 'invalid_authorization_token')
+    assert_failure response
+    assert_match %r{Requested Charge does not exist}, response.message
+  end
+
+  def test_unsuccessful_refund
+    assert purchase = @gateway.purchase(@amount, @credit_card, @options)
+    assert_success purchase
+    assert purchase.authorization
+
+    assert refund = @gateway.refund(@amount + 1, purchase.authorization, @options)
+    assert_failure refund
+    assert_match %r{Wrong Refund data}, refund.message
+  end
+
+  def test_successful_void
+    assert purchase = @gateway.purchase(@amount, @credit_card, @options)
```

Comment:
The primary purpose of `void` in ActiveMerchant is to cancel authorizations - does it work for that? Can you test that scenario, please?